tests/integration/components/task-card-test.js

```diff
@@ -293,3 +300,48 @@ test('assignment dropdown does not render if user has no ability', function(asse
     assert.equal(user.icon.url, 'test.png');
   });
 });
+
+test('the selected-item component is visable when a task has a user assigned ', function(assert) {
+  assert.expect(5);
+
+  let task = { id: 'task' };
+  let user = { id: 'user', username: 'testuser', photoThumbUrl: 'test.png' };
+  let users = [user];
+  let taskUser = user;
+
+  setProperties(this, { task, users, taskUser });
+
+  renderPage();
+  assert.equal(page.selectedItem.selectedIcon.src, taskUser.photoThumbUrl, 'the selected user photo icon renders');
+  assert.notOk(page.tooltip, 'there is no tooltip because mouse is not hovering.');
+  assertTooltipNotRendered(assert);
+
+  page.selectedItem.mouseenter();
+
+  assertTooltipVisible(assert);
+  assert.equal(page.selectedItem.tooltip.text, `Assigned to ${taskUser.username}`, 'the tooltip renders the correct text.');
+});
+
+test('the unselected-item component is visable when a task has a user assigned ', function(assert) {
+  assert.expect(4);
+  let task = { userTask: { isLoading: false }, id: 'task' };
+  let user = { id: 'user', username: 'testuser' };
+  let users = [user];
+
+  setProperties(this, { task, users });
+
+  renderPage();
+  assert.ok(page.unselectedItem.unselectedIcon.isVisible, 'the unselected user icon renders.');
+  assert.notOk(page.tooltip, 'there is no tooltip because mouse is not hovering.');
+  assertTooltipNotRendered(assert);
+
+  page.unselectedItem.mouseenter();
+
+  assertTooltipVisible(assert);
+  assert.equal(page.unselectedItem.tooltip.text, 'Assigned this task', 'the tooltip renders the correct text.');
+
+  page.unselectedItem.mouseleave();
+
+  assertTooltipRendered(assert);
+  assertTooltipNotVisible(assert);
+});
```

Comment:
These two tests are actually testing 4 behaviors, 2 per test.

- one test checks if selected item is rendered when appropriate, and if the deferred tooltip rendering behavior works on it.
- another test checks if the unselected item component is rendered when appropriate and if the deferred tooltip rendering behavior works on it

DRY is a great concept, but generally, even at the cost of repeating ourselves, we want tests that each test one aspect of a component's behavior. At least, that's the approach we want to use with integration and unit tests.

With acceptance tests, since they are much slower and test overall end to end behavior, we want one test per discrete user action and outcome, which then asserts all aspects of it behaved correctly.

In this specific case, that means we'd prefer 4 tests

- check if selected item is rendered when appropriate,
- check if deferred tooltip rendering behavior works on selected item
- check if the unselected item component is rendered when appropriate
- check if the the deferred tooltip rendering behavior works on unselected item

However, from that list, it's clear that tests 2 and 4 are already covered in `selected-item-test` and `unselected-item-test`, so we don't actually need them here. All we need is tests 1 and 3

Specifically, that means we just want to leave the `.isVisible` assertion in each of the two tests, and remove everything else.

That probably annoys you, but you did the work correctly, so it has value, and I'd likely go through the same process myself, before deciding on this route.